tests/functional/context_methods/test_builtin_functions.py

```diff
@@ -105,26 +109,18 @@ def test_builtin_invocation_args_dict_function(self, project):
 
         parsed_logs = parse_json_logs(log_output)
         result = find_result_in_parsed_logs(parsed_logs, "invocation_result")
-
         assert result
-
-        # Result is checked in two parts because profiles_dir is unique each test run
-        expected = "invocation_result: {'debug': True, 'log_format': 'json', 'write_json': True, 'use_colors': True, 'printer_width': 80, 'version_check': True, 'partial_parse': True, 'static_parser': True, 'profiles_dir': "
-        assert expected in str(result)
-
+        # The result should include a dictionary of all flags with default values that aren't None
         expected = (
-            "'send_anonymous_usage_stats': False",
             "'quiet': False",
-            "'no_print': False",
+            "'print': True",
             "'cache_selected_only': False",
             "'macro': 'validate_invocation'",
-            "'args': '{my_variable: test_variable}'",
```

Comment:
This is now a real dictionary, instead of a string-that-can-be-coerced-to-yaml. Better!